lib/web_ui/lib/src/engine/window.dart

```diff
@@ -214,17 +217,30 @@ class EngineWindow extends ui.Window {
     scheduleFrameCallback!();
   }
 
+  @visibleForTesting
+  JsUrlStrategy? urlStrategy = ui.debugEmulateFlutterTesterEnvironment
+      ? null
+      : convertToJsUrlStrategy(const HashUrlStrategy());
+
   /// Change the strategy to use for handling browser history location.
   /// Setting this member will automatically update [_browserHistory].
   ///
   /// By setting this to null, the browser history will be disabled.
-  set locationStrategy(LocationStrategy? strategy) {
-    _browserHistory.setLocationStrategy(strategy);
+  Future<void> setUrlStrategy(JsUrlStrategy? strategy) {
```

Comment:
`assert(!_isHistoryInitialized)`?

Reply:
This can be called again after history has been initialized. It replaces the old URL strategy. At least that's how it works now.

Maybe it doesn't make sense to allow that to happen? We used to allow it in the past because we want the user to be able to override the default one. But now that things are changing and the history is initialized lazily, I actually don't see the point in "replacing" a URL strategy anymore. The assertion would also help catching history tests that aren't cleaning up properly after themselves.

TL;DR: I agree with adding the assertion :)